crates/krun/src/bin/krun.rs

```diff
@@ -62,17 +62,69 @@ fn main() -> Result<()> {
     };
 
     {
+        let cpu_list = if !options.cpu_list.is_empty() {
+            options.cpu_list
+        } else {
+            let mut perf_max_freq = None;
```

Comment:
I think this section of code should be moved to a method in `utils`, and in case anything fails warn and fallback to some safe value (i.e. 4).

Reply:
I'll move it to another file in the `krun` crate.